app/models/location.rb

```diff
@@ -0,0 +1,21 @@
+class Location
+  include Mongoid::Document
+  include Mongoid::Timestamps
+
+  field :longlat, type: Hash
+  field :city, type: String
+  field :region, type: String
+
+  has_many :attractions
+
+  validates_presence_of :city, :region, :longlat
+
+  def latitude
```

Comment:
Since both `Attraction` and `Location` have `latitude` and `longitude` methods, this is the perfect (and really only case) for using a `concern.`

You might have `app/models/concerns/long_lat.rb`
```ruby
module LongLat
  def latitude
    longlat['coordinates'][1]
  end

  ...
end
```

Then in `Attraction` and `Locations` you'd just `include LongLat`